modules/tinymce/src/core/main/ts/fmt/FormatUtils.ts

```diff
@@ -169,6 +170,59 @@ const getParents = function (dom: DOMUtils, node: Node, selector?: string) {
   return dom.getParents(node, selector, dom.getRoot());
 };
 
+const isVariableFormatName = (editor: Editor, formatName: string): boolean => {
+  const hasVariableValues = (format: Format) => {
+    const isVariableValue = (val: string): boolean => val.length > 1 && val.charAt(0) === '%';
+    const stylesFieldHas = Obj.get(format, 'styles').exists((stylesField) => Arr.exists(Obj.values(stylesField), isVariableValue));
+    const attributesFieldHas = Obj.get(format, 'attributes').exists((attributesField) => Arr.exists(Obj.values(attributesField), isVariableValue));
+    return stylesFieldHas || attributesFieldHas;
+  };
+  return Arr.exists(editor.formatter.get(formatName) as Format[], (format) => hasVariableValues(format));
+};
+
+/**
+ *  Get all of the format names present on the specifed node
+ */
+const getFormatNamesPresent = (editor: Editor, node: Node, blacklist: string[] = [ 'removeformat' ], deep: boolean = false): string[] => {
```

Comment:
This seems as though it should be something in `MatchFormat` instead of `FormatUtils`, as it's attempting to find out what formats match.

Reply:
Also instead of returning names, I have to wonder if this would be better returning matched formats. As there's a lot of additional calls being done to lookup the format based on the names when it looks like it could be avoided.